hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -140,6 +140,7 @@ public class MapConfig implements IdentifiedDataSerializable, NamedConfig {
             .setMaxSizePolicy(DEFAULT_MAX_SIZE_POLICY)
             .setSize(DEFAULT_MAX_SIZE);
     private TieredStoreConfig tieredStoreConfig = new TieredStoreConfig();
+    private List<PartitioningAttributeConfig> partitioningAttributeConfigs;
```

Comment:
why not add `List<PartitioningAttributeConfig>` to `PartitioningStrategyConfig`? It would be more intuitive and resemble original syntax: `<partition-strategy>com.hazelcast.partition.strategy.AttributePartitioningStrategy(myAttribute1, myAttribute2, …)</partition-strategy>`. BTW, maybe we should update the docs?

Reply:
but I think that explicit list is better than parsing text using regex

Reply:
IIRC its a bit complicated to do that, but if you have a quick solution, please feel free to push it.

Reply:
why? is there a problem with client protocol or backward compatibility? MapConfig is not returned to client.

BTW, should we support `PartitioningAttributeConfig` in dynamic config (client protocol change needed in `addMapConfig`)?

Reply:
TL;DR is that I'd need to change how the driver works - it currently processes top level nodes sequentially so its not possible to (efficiently) combine processing of two configuration nodes. 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java#L277

There's an ugly solution of create-if-not-exists for the structure itself I guess, but I'd say its not worth it over having this for now.

Reply:
As far as dynamic config goes, I was wondering that too, I'm not certain but think MapConfig doesn't require any special handling aside from the currently present one.

Reply:
ok, thanks for explanation of the place where we can add the attributes. It looks like a serious limitation to add new fields to existing config.

In order to fully support dynamic config, new fields need to be added to https://github.com/hazelcast/hazelcast-client-protocol/blob/master/protocol-definitions/DynamicConfig.yaml#L706 (and obviously need to handled on server side)

Reply:
As discuess with @viliam-durina, we plan to implement support for attribute based partitioning strategy in dynamic config on client in separate PR.